app/src/main/java/com/rockthevote/grommet/ui/eventFlow/SessionTimeTracking.java

```diff
@@ -164,17 +158,8 @@ public void onClickEdit(View v) {
 
     @OnClick(R.id.clock_in_button)
     public void onClickClockIn(View button) {
-
         if (button.isSelected()) {
-            new AlertDialog.Builder(getContext())
-                    .setMessage(R.string.clock_out_dialog_text)
-                    .setPositiveButton(R.string.dialog_yes, (dialogInterface, i) -> {
-                        clockOut();
-                    })
-                    .setNegativeButton(R.string.dialog_no,
-                            (dialogInterface, i) -> dialogInterface.dismiss())
-                    .create()
-                    .show();
+            viewModel.asyncCanClockOut(this::canClockOut, this::canNotClockOut);
```

Comment:
this is clever

Reply:
you're passing java lambda's to a kotlin higher order function, right? Didn't know you could do this.

I assume that's what the `Kotlin.Unit.Instance` lets you do, is it something similar to annotations?